### PR TITLE
Adding Payment Integration to Registration

### PR DESCRIPTION
**Pull request by EiffL**
This PR adds an integration with the Chicago payment website https://uchicago.service-now.com/it?id=kb_article&kb=KB06000437

If the user sends all relevant data, the heroku server will log that info, and then send back the info needed to proceed to payment, which will redirect the user to the payment portal.

2 more things to add:
- [x] only proceed with payment when in-person
- [x] (optional) add a callback to the heroku server to log the sucessfull payment in the registration database

A demo is available [here](https://raw.githack.com/LSSTDESC/meeting-registration-form/august22dev/index.html?backend=https://desc-meeting-august22-dev.herokuapp.com&secret=fc04944077a3af7236b18bebb0d8994227e0f7f7bcf132eae3d4d1b6b1714187). But **careful, the payment website is live**, it will charge your credit card if you go through the process

### CODE REVIEW

**Comment by EiffL** [comment]
Everything should be ready to review @JoanneBogart 

A live demo is available at [this page](https://raw.githack.com/LSSTDESC/meeting-registration-form/august22dev/index.html?backend=https://desc-meeting-august22-dev.herokuapp.com&secret=fc04944077a3af7236b18bebb0d8994227e0f7f7bcf132eae3d4d1b6b1714187)

File: templates/success.html [hunk 1]
```diff
@@ -14,14 +15,16 @@
...
 
     <br>
-    The <a href="https://confluence.slac.stanford.edu/x/fQzUEw" target="_blank">Participants tab</a> on the confluence page links to a live participant list, where you can confirm your
+    The <a href="https://confluence.slac.stanford.edu/x/fQzUEw" target="_blank">Participants tab</a> on the confluence
+    page links to a live participant list, where you can confirm your
     registration.
     <br>
     <br>
-    And have a look at the <a href="https://confluence.slac.stanford.edu/x/DwzUEw" target="_blank">agenda page</a> to check the program of our virtual meeting!
+    And have a look at the <a href="https://confluence.slac.stanford.edu/x/DwzUEw" target="_blank">agenda page</a> to
```
Comment (JoanneBogart) [inline templates/success.html:26]: Link should be changed to 
https://confluence.slac.stanford.edu/x/ooXDF
(though page has nothing useful in it yet)

File: templates/success.html [hunk 2]
```diff
@@ -14,14 +15,16 @@
     <h2>You are registered for the DESC August 2022 collaboration meeting!</h2>
     <br>
     <br>
-    {{data.first_name}}, we are looking forward to seeing you at our virtual meeting!<br>
+    {{data.first_name}}, we are looking forward to seeing you at the meeting!<br>
 
     <br>
-    The <a href="https://confluence.slac.stanford.edu/x/fQzUEw" target="_blank">Participants tab</a> on the confluence page links to a live participant list, where you can confirm your
+    The <a href="https://confluence.slac.stanford.edu/x/fQzUEw" target="_blank">Participants tab</a> on the confluence
```
Comment (JoanneBogart) [inline templates/success.html:21]: Link should be changed to
https://confluence.slac.stanford.edu/x/w4XDF
(though page has nothing useful in it yet)

File: templates/payment_success.html [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+        We are looking forward to seeing you at the meeting!<br>
+
+        <br>
+        The <a href="https://confluence.slac.stanford.edu/x/fQzUEw" target="_blank">Participants tab</a> on the
+        confluence page links to a live participant list, where you can confirm your
+        registration.
+        <br>
+        <br>
+        And have a look at the <a href="https://confluence.slac.stanford.edu/x/DwzUEw" target="_blank">agenda page</a>
```
Comment (JoanneBogart) [inline templates/payment_success.html:34]: Change link to point to Aug 2022 agenda page.  See value in comment below.

File: templates/payment_success.html [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+            might notice that <b>the receipt will mention CMB Pol Workshop but this is normal</b>. If you have any
+            questions
+            regarding payment, please reach out to the LOC.
+
+        </div>
+
+        We are looking forward to seeing you at the meeting!<br>
+
+        <br>
+        The <a href="https://confluence.slac.stanford.edu/x/fQzUEw" target="_blank">Participants tab</a> on the
```
Comment (JoanneBogart) [inline templates/payment_success.html:29]: Change link to point to Aug 2022 participants page.   See value in comment below

File: templates/payment_success.html [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+    <div class="container">
+        <h2>You are registered for the DESC August 2022 collaboration meeting!</h2>
+        <br>
+
+        <div class="alert alert-info" role="alert">
+            You should have received by email a confirmation of your payment from <i>maureenlowery@uchicago.edu</i>. You
+            might notice that <b>the receipt will mention CMB Pol Workshop but this is normal</b>. If you have any
+            questions
+            regarding payment, please reach out to the LOC.
```
Comment (JoanneBogart) [inline templates/payment_success.html:22]: Can we be more explicit about how to reach out to LOC?  Email? Slack channel?

File: index.html [hunk 6]
```diff
@@ -128,148 +136,233 @@
...
+    There is an in-person registration fee for this meeting. Remote
     participation is free.
     <br>
-    Please join the LSSTC Slack channel <a href="https://lsstc.slack.com/archives/C90AMAG80" target="_blank">#desc-collab-meeting</a> to keep updated on chats, discussions, and announcements during the meeting.
+    Please join the LSSTC Slack channel <a href="https://lsstc.slack.com/archives/C90AMAG80"
+      target="_blank">#desc-collab-meeting</a> to keep updated on chats, discussions, and announcements during the
+    meeting.
     <br>
-    You can find the the details of the schedule on this <a href="https://confluence.slac.stanford.edu/x/agvUEw" target="_blank">confluence page</a>.
+    You can find the the details of the schedule on this <a href="https://confluence.slac.stanford.edu/x/agvUEw"
```
Comment (JoanneBogart) [inline index.html:146]: This currently brings up the home page for the Feb 2022  Meeting.  
URL for new home page is 
https://confluence.slac.stanford.edu/x/CZh8F

URL for new (currently empty) agenda page is
https://confluence.slac.stanford.edu/x/ooXDF

File: index.html [hunk 7]
```diff
@@ -128,148 +136,233 @@
...
+
+
+        <div class="container">
+
+          <h5>Covid-19 regulations</h5>
+
+          <p>To ensure the safety of the meeting, in-person participants are asked
+            to verify that they have received a full course of vaccination and that
+            they will comply with the regulations for Covid-19 safety at the event,
+            which includes wearing a mask during in-door sessions.</p>
```
Comment (JoanneBogart) [inline index.html:254]: Change "in-door" to "indoor"

**Review by JoanneBogart — CHANGES_REQUESTED** [review]
In addition to the URL updates needed, it seems that there is a hang-up with registering for remote participation.  Otherwise it looks good.

File: templates/payment_success.html (hunk 5 above)
Comment (EiffL) [reply templates/payment_success.html:21]: added link to hotline channel, confluence is down so can't find link to LOC

**Pull request merged**